types/beyonk__gdpr-cookie-consent-banner/index.d.ts

```diff
@@ -0,0 +1,10 @@
+// Type definitions for @beyonk/gdpr-cookie-consent-banner 9.0
+// Project: https://github.com/beyonk-adventures/gdpr-cookie-consent-banner
+// Definitions by: Ben Grynhaus <https://github.com/bengry>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference path="./dist/beyonk__gdpr-cookie-consent-banner.umd.d.ts" />
+
+import attachBanner from './dist/beyonk__gdpr-cookie-consent-banner.esm';
+
+export default attachBanner;
```

Comment:
```suggestion
export default attachBanner;

export as namespace GdprConsent;
```

Reply:
that should be enough. One should avoid to create files that does not exists in native JS mdoule (e.g. there is no 'umd.*' file in the bundle). You can retract the changes for exports (package.json + umd bundle, etc)

Reply:
With this suggested change in the UMD mode I get a `GdprConsent` global with a function named `default` instead of `attachBanner`. That was my initial issue and why I created several files.

![CleanShot 2022-10-08 at 10 37 08@2x](https://user-images.githubusercontent.com/260431/194696007-a1f37d86-d91f-412e-9cb4-86b41f6fa2b6.png)

Reply:
@bengry well, I was wrong, but I believe becuase the JS native module is complex one, 
to support both:
```ts
/// <reference types="@beyonk/gdpr-cookie-consent-banner/" />

GdprConsent.attachBanner(document.body)
```
and:
```ts
import attachBanner from "@beyonk/gdpr-cookie-consent-banner";

attachBanner(document.body)
```
or
```ts
import attachBanner from "@beyonk/gdpr-cookie-consent-banner/dist/esm/bundle.js";

attachBanner(document.body)
```
can you try?:
```ts
// Type definitions for @beyonk/gdpr-cookie-consent-banner 9.0
// Project: https://github.com/beyonk-adventures/gdpr-cookie-consent-banner
// Definitions by: Ben Grynhaus <https://github.com/bengry>
// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped

interface Choice {
    label: string;
    description: string;
    value: boolean;
}

type Category = 'necessary' | 'tracking' | 'analytics' | 'marketing';

interface Options {
    cookieName?: string;
    cookieConfig?: {
        domain: string;
        path: string;
    };
    heading?: string;
    description?: string;
    acceptLabel?: string;
    rejectLabel?: string;
    settingsLabel?: string;
    closeLabel?: string;
    choices?: Partial<Record<Category, Choice | false>>;
    showEditIcon?: boolean;
    categories?: Partial<Record<Category, () => void>>;
}

interface GdprConsentInterface {
    attachBanner(bodyElement: HTMLElement, options?: Options): void;
}

declare var GdprConsent: GdprConsentInterface;

declare module '@beyonk/gdpr-cookie-consent-banner' {
    const attachBanner: GdprConsentInterface['attachBanner'];
    export default attachBanner;
}

declare module '@beyonk/gdpr-cookie-consent-banner/dist/esm/bundle.js' {
    const attachBanner: GdprConsentInterface['attachBanner'];
    export default attachBanner;
}

declare module '@beyonk/gdpr-cookie-consent-banner/dist/style.css' {}
```

if it does not work, stick only to ESM module and skip global script support
(in the sample code .d.ts yu would eventually have to add eslint rule exclusions as this is a single file multimodule export)

Reply:
Thanks @peterblazejewicz! Looks like it's working as expected. I changed the `var` for `const`, but all the tests pass without further changes.

Much simpler than my original approach of multiple files.